python/dolfinx/mesh.py

```diff
@@ -47,17 +52,31 @@ def __init__(self, comm: _MPI.Comm, topology: _cpp.mesh.Topology,
             Mesh objects are not generally created using this class directly.
 
         """
-        super().__init__(comm, topology, geometry)
+        # super().__init__(comm, topology, geometry)
+        self._mesh = mesh
```

Comment:
If I understand correctly, is this the C++ mesh object? If so, you may want to consider calling it `_cpp_object`, since  that convention is used several times elsewhere, e.g. in `python/dolfinx/fem/function.py` and `python/dolfinx/fem/dofmap.py`

Reply:
@francesco-ballarin yes, can do.

Caveat: I used `_cpp_object` at the very when moving DOFLIN-old to pybind11. Are we happy with the name?

Reply:
I personally am quite happy wit that name :)